src/Core/Config/Config.php

```diff
@@ -119,4 +118,23 @@ public static function forClass($class)
     {
         return new Config_ForClass($class);
     }
+
+    /**
+     * Perform the given operation in an isolated config state.
+     * On return, the config state will be restored, so any modifications are temporary.
+     *
+     * @param callable $callback Callback to run. Will be passed the nested config state as a parameter
+     * @return mixed Result of callback
+     */
+    public static function withConfig($callback)
+    {
+        static::nest();
```

Comment:
You don't need to explicitly nest the config if you're calling modify() straight afterwards, that happens internally already

Reply:
I thought it only happens if it's not mutable? https://github.com/silverstripe/silverstripe-framework/blob/4/src/Core/Config/Config.php#L58

Reply:
The first time you call `::modify()` it returns a mutable interface. It will nest if it needs to - that code is a little confusing to read but it's designed to nest when you mutate, which is why we advise against modifying config in PHP and using YAML instead to avoid this performance hit

Reply:
The impact of having both of these lines here is minimal, it's like one extra if statement being processed in the code you've referenced, but it doesn't need to be there either way

Reply:
I had to add this back as it's breaking the unit tests. I think this means it's needed.

Reply:
Ok, sorry for the noise